client/purplecaffeine/core.py

```diff
@@ -67,18 +67,40 @@ def __init__(
             tags (List[str]): list of tags in string format
         """
         self.uuid = uuid or str(uuid4())
-        self.name = name
-        self.storage = storage or LocalStorage(path="./")
-
-        self.description = description or ""
-        self.metrics = metrics or []
-        self.parameters = parameters or []
-        self.circuits = circuits or []
-        self.operators = operators or []
-        self.artifacts = artifacts or []
-        self.texts = texts or []
-        self.arrays = arrays or []
-        self.tags = tags or []
+        self.name = name or os.environ.get("PURPLE_CAFFEINE_TRIAL_NAME")
+        if self.name is None:
+            raise PurpleCaffeineException(
+                "Please specify name of trial or configure it using env variables"
+            )
+
+        if storage is None:
+            storage_type = os.environ.get("PURPLE_CAFFEINE_STORAGE_CLASS", "LocalStorage")
+            storage_mapping: Dict[str, BaseStorage] = {
+                "LocalStorage": LocalStorage,
+                "S3Storage": S3Storage,
+                "ApiStorage": ApiStorage
+            }
+            self.storage = storage_mapping.get(storage_type)()
+        else:
+            self.storage = storage
+
+        self.description = description or os.environ.get("PURPLE_CAFFEINE_TRIAL_DESCRIPTION", "")
+        self.metrics = metrics or os.environ.get("PURPLE_CAFFEINE_TRIAL_METRICS", "").split(",")
```

Comment:
I do not think we need metrics, parameters, circuit, operators, artifacts, texts, array to be from env vars.

Everything else is good!

Reply:
What about `tags`? should it be allowed to be taken from env variables?

Reply:
yes, tags should be possible to set from env